packages/event-bus-redis/src/services/event-bus-redis.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  AbstractEventBusModuleService,
+  ConfigModule,
+  ConfigurableModuleDeclaration,
+  Logger,
+  MODULE_RESOURCE_TYPE
+} from "@medusajs/medusa"
+import { Queue, Worker } from "bullmq"
+import Redis from "ioredis"
+import { MedusaError } from "medusa-core-utils"
+import { BullJob, EmitOptions, EventBusRedisModuleOptions } from "../types"
+
+type InjectedDependencies = {
+  logger: Logger
+  configModule: ConfigModule
+  redisConnection: Redis
+}
+
+const COMPLETED_JOB_TTL = 10000
+
+/**
+ * Can keep track of multiple subscribers to different events and run the
+ * subscribers when events happen. Events will run asynchronously.
+ */
+export default class RedisEventBusService extends AbstractEventBusModuleService {
+  protected readonly config_: ConfigModule
+  protected readonly logger_: Logger
+  protected readonly moduleOptions_: EventBusRedisModuleOptions
+  protected readonly moduleDeclaration_: ConfigurableModuleDeclaration
+
+  protected queue_: Queue
+
+  constructor(
+    { configModule, logger, redisConnection }: InjectedDependencies,
+    moduleOptions: EventBusRedisModuleOptions = {},
+    moduleDeclaration: ConfigurableModuleDeclaration
+  ) {
+    // @ts-ignore
+    super(...arguments)
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+
+    this.moduleOptions_ = moduleOptions
+    this.moduleDeclaration_ = moduleDeclaration
+    this.config_ = configModule
+    this.logger_ = logger
+
+    this.queue_ = new Queue(this.moduleOptions_.queueName ?? `events-queue`, {
+      prefix: `${this.constructor.name}`,
+      ...(this.moduleOptions_.queueOptions ?? {}),
+      connection: redisConnection,
+    }).on("error", (e: any) => {
+      logger.error(`An error occurred with the BullMQ Queue: ${e.message}`)
```

Comment:
suggestion: I think to make the errors cleaner we should include an EOL before the error for all logs